tensorflow/compiler/mlir/tosa/transforms/fuse_bias_tf.cc

```diff
@@ -68,32 +68,28 @@ LogicalResult ConvertTFBiasAddOp::matchAndRewrite(
 
   auto output_type =
       tf_biasadd_op.getResult().getType().dyn_cast<RankedTensorType>();
+
   // Not a ranked tensor output
   if (!output_type) return failure();
 
   auto value = tf_biasadd_op.value();
   auto bias = tf_biasadd_op.bias();
+  auto bias_shape = bias.getType().dyn_cast<RankedTensorType>().getShape();
+  // Bias tensor that feeds into tosa.conv2d or tosa.conv3d must be rank 1
+  if (bias_shape.size() != 1) return failure();
 
   TF::Conv2DOp tf_conv2d_op =
       dyn_cast_or_null<TF::Conv2DOp>(value.getDefiningOp());
 
-  if (!tf_conv2d_op) {
-    return failure();
-  }
-
+  if (tf_conv2d_op) {
```

Comment:
Nit: to reduce scope/life range/avoid accidental dereferencing when null, prefer

if (TF::Conv2DOp tf_conv2d_op = ...) { ... }

Reply:
Fair enough. Done.